core/src/Revolution/Processors/Element/Create.php

```diff
@@ -50,13 +58,34 @@ public function cleanup()
      */
     public function beforeSave()
     {
-        $name = $this->getProperty('name');
+        $locked = (bool)$this->getProperty('locked', false);
+        $this->object->set('locked', $locked);
+
+        $elementClassName = array_pop(explode('\\', $this->classKey));
+
+        if ($elementClassName === 'modTemplateVar') {
+            if ($caption = $this->getProperty('caption', '')) {
+                $this->object->set('caption', strip_tags($caption));
+            }
```

Comment:
1. It could be written shorter.

2. Why do you prefer assignment within a condition? 

`> Although the programmer's intent could have been to assign b to a and test the value of the result, this usage frequently occurs when the programmer mistakenly used the assignment operator = rather than the equality operator ==.`
(more details: https://wiki.sei.cmu.edu/confluence/display/java/EXP51-J.+Do+not+perform+assignments+in+conditional+expressions)

```suggestion
            $this->object->set('caption', strip_tags($this->getProperty('caption', '')));
```

Reply:
This is done this way because captions only apply to TVs and that property will not be present in the other elements.

Reply:
Checking a specific field (`caption`) to determine the entity class (TV in this case) is an unreliable construct. Explicit is better than implicit.
Also this does not answer the question about assignment within a condition.

Reply:
A couple things:
1. To answer your question: I often make an assignment in a condition when the variable is only needed within that condition. I think that should be self-evident. It's one less line and just as easy to understand.
2. I'm not trying to determine the entity class, I was just pointing out that `caption` is not a property of every element (it's currently only a property of TVs).

Reply:
Do you even need the variable `$caption`?